### PR TITLE
Grpc tracing - Trace Context

### PR DESCRIPTION
**Pull request by Vizerai**
Recreated pull request with correct user/branch.

Initial submission with trace context files.

### CODE REVIEW

File: src/core/ext/census/trace_context.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+// This function assumes the TraceContext is valid.
+size_t encode_trace_context(google_trace_TraceContext *ctxt, uint8_t *buffer,
+                            const size_t buf_size) {
+  // Create a stream that will write to our buffer.
+  pb_ostream_t stream = pb_ostream_from_buffer(buffer, buf_size);
+
+  // encode message
+  bool status = pb_encode(&stream, google_trace_TraceContext_fields, ctxt);
+  size_t nbytes = stream.bytes_written;
```
Comment (a-veitch) [inline src/core/ext/census/trace_context.c]: Suggestion: code after pb_encode could be:

if (!status) {
  gpr_log(...);
  return 0;
}
return stream.bytes_written;

File: src/core/ext/census/trace_context.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  // Create a stream that reads nbytes from the buffer.
+  pb_istream_t stream = pb_istream_from_buffer(buffer, nbytes);
+
+  // decode message
+  bool status = pb_decode(&stream, google_trace_TraceContext_fields, ctxt);
+
+  if (!status) {
+    gpr_log(GPR_DEBUG, "TraceContext decoding failed: %s",
+            PB_GET_ERROR(&stream));
+  }
```
Comment (a-veitch) [inline src/core/ext/census/trace_context.c]: Either 'return false' here, or put the other checks in an else (and nest them as well - no need to check span_id if trace_id check fails...). My preference would be to return directly (after each case) and just have 'return true' at the end.

File: test/core/census/trace_context_test.c [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+int main(int argc, char **argv) {
+  grpc_test_init(argc, argv);
+  test_full();
+  test_empty();
+  test_trace_only();
+  test_span_only();
+  test_encode_decode();
+  test_corrupt();
+  test_no_sample();
+
```
Comment (a-veitch) [inline test/core/census/trace_context_test.c]: Add a test that tries to encode to a too-small buffer, and check that it fails.

File: src/core/ext/census/trace_context.c (hunk 1 above)
Comment (Vizerai) [inline src/core/ext/census/trace_context.c]: This has been changed.

File: src/core/ext/census/trace_context.c (hunk 2 above)
Comment (Vizerai) [inline src/core/ext/census/trace_context.c]: It directly returns false in those conditions now, and returns true at the end of the function.

File: test/core/census/trace_context_test.c (hunk 3 above)
Comment (Vizerai) [inline test/core/census/trace_context_test.c]: Added

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Pull request closed without merging**